### PR TITLE
Add AES Key Wrap Algorithm

### PR DESCRIPTION
**Pull request by sylvainpelissier**
**Checklist**

- [ ] Closing issues: #issue
- [x] Mark this if you consider it ready to merge
- [x] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**
Implementation of AES Key Wrap Algorithm according to [RFC 3394](https://datatracker.ietf.org/doc/html/rfc3394) in radare2.

### CODE REVIEW

File: libr/crypto/p/crypto_aes_wrap.c [hunk 1]
```diff
@@ -0,0 +1,149 @@
+/* radare - LGPL - Copyright 2021 - Sylvain Pelissier 
+   Implementation of AES Key Wrap Algorithm (RFC 3394) */
+
+#include <r_lib.h>
+#include <r_crypto.h>
+#include "crypto_aes_algo.h"
+
+#define BLOCK_SIZE 8
+
+static struct aes_state st;
```
Comment (trufae) [inline libr/crypto/p/crypto_aes_wrap.c:10]: may be good to not use globals. i think crypto plugins should have an init method to return that void* state to be used

**Review by trufae — CHANGES_REQUESTED** [review]

File: libr/crypto/p/crypto_aes_wrap.c (hunk 1 above)
Comment (sylvainpelissier) [reply libr/crypto/p/crypto_aes_wrap.c:10]: I do not see what is the problem with global here ? Should I change `RCryptoPlugin` to add a init function ?

Comment (trufae) [reply libr/crypto/p/crypto_aes_wrap.c:10]: The problem would be when this code gets used from two different threads or coroutines. Also another big problem with globals is that on some OS like android they are loaded once across different executions which means that the initial global state is initialized once, the subsequent executions will missbehave

Comment (sylvainpelissier) [reply libr/crypto/p/crypto_aes_wrap.c:10]: Since it require modification of `RCryptoPlugin` struct I would prefer to handle that in a separate PR.

Comment (trufae) [reply libr/crypto/p/crypto_aes_wrap.c:10]: 👍

**Pull request merged**